### ISSUE TITLE
[bug] meta: extract any remaining issues from the pst quest list

### ISSUE THREAD

**Issue by lynxlynxlynx** [gemrb/gemrb#439]
Here's chiv detailed list from 2015:
https://github.com/gemrb/gemrb/wiki/PST-Quests

It still contains some complaints, stuff marked RESEARCH, RETEST and maybe more. Any such thing needs a separate issue or just a note that it has been resolved.

**Comment by chilvence** [comment]
I haven't made it to the quest list yet, but I have got halfway through the general bug list and started tracker issues for unresolved bugs.

**Comment by chilvence** [comment]
I will be working through the Quests List again soon. Apart from the two issues I hit the other week, I haven't found any regressions or errors so far.  Most of the problems appear in the late stage of the game, it's just taking time to get there. I am also a bit setback because I managed to corrupt the sd card on a friends odroid console, so now I am having to learn how to set that up...which would normally be a fun project, except I really wasn't planning on it right now.

**Comment by lynxlynxlynx** [comment]
There's no rush, but good to hear much of the problems are stale and fixed.

**Comment by chilvence** [comment]
There is a note in that page about APP_BUDDY behaviour, which I don't think I ever got around to asking about, I wasn't sure what I should be checking for in the original game vs gemrb . I can't scrub that one off the list because I am not sure what to write an issue about. 

Most other issues on that page so far have melted away thanks to your bug-fixing rampage :)

Although I still haven't hit the point where all the 'retest' bits are, I am feeling optimistic about it.

**Comment by lynxlynxlynx** [comment]
Cool, thanks. I'll port the buddy thing, since I added it.

**Comment by chilvence** [comment]
I have made it to Ravel's maze. So far I have been able to accomplish everything I have set out to do other than what has already been noted. The only new snag I found was that after talking to Yi'miin in ar0500, he offers to help you , starts a cutscene that leads you into an alley, and then you get stuck forever. I will open an issue with a save game shortly . I also ran into another annoyance in Vrischika's store where everyone would turn hostile, but I was not able to figure out the cause or reproduce it after backtracking from an earlier save, so I don't know what to report :/

All that remains to try is the final third of the game, and I think that since some related problems have already been resolved, it will be a lot less error prone than before.

**Comment by chilvence** [comment]
Now I am at the final area of the fortress of regret. Until this area, I encountered no problems that are not already known/reported. 

There were a couple of similar instances to #573 where enemy characters don't seem to support each other, one in curst prison inside the entrance to the rings, and another in curst carceri version where a mage commands his thokola slavers to attack and they do nothing. I am 99% sure they are related to the same problem, and it could be that it has been a problem through the whole game but there were no scenarios that made it obvious there was an issue.

As for the rest of the bugs on that list, they have mostly gone away thanks to general progress on Gemrb or were duplicate symptoms of already fixed bugs. There were some quests where I no longer remember the original issue because there is no page history, but I did not notice any problems while doing them. Being separated in the Fortress now works correctly. I no longer get stuck in the area where Ignus confronts you. The reunion with your incarnations seems to work pretty much fine. The only thing that has stopped me in my tracks now is the final meeting the last boss cutscene.

The last thing on the quest page, the Modron maze, we already know is not fully implemented. So this quests page could now be deleted -- I don't think it will be of any use anymore even if new problems come to light with the main game path, using the bug tracker makes much more sense.

Of course, the good news is that since I didn't find any more major bugs on this play through, I have a whole desk full of notes about minor bugs that need attention...

**Comment by lynxlynxlynx** [comment]
Please report what exactly is wrong with the maze, separately. We have some confusing code, so it's not clearly if an extra seed is needed or not (the easy one shortcircuits a function).

Smaller stuff and the final cutscene also deserve their own reports. I also remember something from the alley giving birth cutscene: shouldn't the stone face animation only play once? It looked a bit off to me, but I didn't compare with the original.

Things remaining on the page without your feedback / needing a report:
> - Deionarra appearance is glitchy this is a minor bug, but her appearance/disappearance animations don't play which rather spoils the moment...
> - Mausoleum Skeletons seem extremely tough compared to stock PST
> - The door at the south east (to the drowned dead, with Barr) should be locked after you get through
>   - RESEARCH, does returning from the next area immediately show them closed already? If not, what do you have to do in the original ... how to reproduce
> - Find a skull of great value: It is possible to do this quest but it doesn't quite happen the same as in the original
> - Kill a companion for an additional power: Not able get grimoire to acknowledge me murdering companions 
> - Split Pestle and Kilnn: GOOD as of 12/2019; A minor GFX issue, he is missing his colour cycle fx for some reason, probably a more generic problem
> - The pillar can take wrong tributes (NPCs etc) for payment
>   - RESEARCH concrete examples needed, testing showed no problems
> - the saved villagers near the gehereleths disappear instantly when you talk to them instead of walking away
> - Ignus does not talk, just goes straight for the kill
> - Clicking on it tells you to kill ignus, even when he's dead....

**Comment by chilvence** [comment]
Apologies, I meant to double back and add comment on most of these, but I guess I was overly relieved I made it to the end of the game. Simultaneously to this, I am also trying to finally 100%  complete Ocarina of Time, which is taking up way more time than I thought it would. I never had a Nintendo 64, so it needs to be done before I can die happy :)

Rather than edit the page again I will just respond to them here:

1 is related to unimplemented animations I am sure, which I would guess falls under #175 so I did not open a new issue

2 no longer seems to be an issue, the skeletons did not to be overpowered, they were nearly unbeatable the last time I played, but I was able to defeat them quite easily even not being a fighter this time.

4 was actually a false report, I misremembered how Morte rejoins the party in the original because it seems strangely implemented, but it is actually happening as intended and I should have deleted that comment but I missed it. Basically, you have to go back to the lower ward to get morte back in your party because that is where he was removed, even if in game it is implied that you were _only just talking to him on a shelf in the basement of an abandoned building_

5 I believe is hard coded, I am still trying to get acclimatized with ida but there is a global 'PM_Killed' which is checked by the script which only seems to be referenced in the exe. I was rolling my eyes at this point because it seems a wasted effort to nag about a bug in a stupidly silly 'ooh look at me I am soo eeeevil I will kill my friends' quest that was hardcoded in the original game and that no one in their right mind would complete anyway, frankly I considered there to be much more important issues to worry about and moved on, but I can add this to the tracker

6 I went to compare this to the the EE version, and it behaves the exact same way. I could have sworn for the life of me that there was originally some palette swapping effect that made the 'unsplit' Pestle  character cycle colours, but obviously it wasn't important enough to preserve. I am guessing that whatever paradigm shifts in graphics rendering that have happened since 1998 have made this a headache to implement. It wasn't exactly pretty anyway, and has no effect on actual gameplay. Please worry about this one after the apocalypse has consumed all life on earth and replaced it with sentient pocket lint.

7 I wasn't able to reproduce, I tried all sorts of different ways to betray my companions but they all seemed to play out as expected. If you try and shove any of them in the pillar, Grace disappears angrily, Annah becomes violent and attacks you, Morte protests and suggests anything to save himself, but you can ignore it and shove him in anyway, Vhailor goes full justice mode etc. It has been a long time but however this part failed on me before does not seem to still be relevant.

8 Isn't exactly a bug, I think it's a difference in the way gemrb and the infinity engine implements EscapeArea(). Just from memory, even actors in BG do this weird quirky thing where they wander about for the odd second or two before vanishing. It doesn't always look that smart in the original game anyway. It doesn't actually negatively affect the game as far as I can tell, this action is used so liberally that I am certain someone  would have noticed ill effects before, but I can add an issue if you reckon it might be important.

9 and 10, the area where you confront Ignus finally seemed to be ok this time. Before, he wouldn't start his dialogue, and the crystal you have to investigate would not respond to his death, but both of those things worked this time around.

As for number 3 on that list.. HMPH. For such a simple issue that should be straightforward, Every time I try to figure it out, I am no wiser. It maybe does deserve its own bug report, but it is infuriating because it actually has no negative effects on the game itself, and I can't find where exactly in the chain of scripts actions and global variables it breaks down. It's a complicated bug that has no consequence whatsoever, besides making the game a bit more lenient. It will need fixing eventually, but to be honest I have been willfully ignoring it because there doesn't seem to be any particular reward for chasing it down. 
 
I do realise that my reflex to ignore particular things probably isn't entirely helpful in the long run, but when I originally wrote this quests page, it was only to identify problems that completely tanked the game, knowing that everything else was just a matter of refinement. In my mind, there was no point in chasing minor issues until the major ones were resolved. I have to remind myself from time to time that the major issues are now resolved and it is ok to think about the minor issues.

**Comment by lynxlynxlynx** [comment]
Thanks, I've updated my comment, since it's more concise. Please work through it once you get the time.

**Comment by lynxlynxlynx** [comment]
I've reported the last things as #615 and with that this big task is over. Many thanks! :cake: 

Of course the other minor issues on your new list are also welcome.

**Comment by chilvence** [comment]
Regarding my list of 'new' issues, it's probably about 20 -30 things I wrote down briefly for further inspection later on, but it would probably take me a long time to investigate and add them one by one. Are you happy for me to just add them all to the tracker so that they are at least known issues?

**Comment by lynxlynxlynx** [comment]
I'd still prefer one per item, since otherwise the discussion will be all over the place and hard to follow. Discipline is hard. But there's also no need to do it all at once, you can take as much time as you want. A handful of actionable items is much better than a long list of potentials with many unknowns.

**Comment by chilvence** [comment]
Oh yeah of course 1 report per item now that we have the tracker. What I mean is, even if some of them might be trivial to fix, I have done no research of my own yet on how to fix them or where the code breaks down, so I can only report them in the most basic 'expected behaviour/actual behaviour/steps to reproduce way' , I won't be able to suggest any solutions

**Comment by lynxlynxlynx** [comment]
Ah, that's fine. :)
